core/types/receipt.go

```diff
@@ -590,6 +594,13 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 			rs[i].FeeScalar = gasParams.feeScalar
 			rs[i].L1BaseFeeScalar = u32ptrTou64ptr(gasParams.l1BaseFeeScalar)
 			rs[i].L1BlobBaseFeeScalar = u32ptrTou64ptr(gasParams.l1BlobBaseFeeScalar)
+			if gasParams.operatorFeeScalar != nil && gasParams.operatorFeeConstant != nil && *gasParams.operatorFeeScalar != 0 && *gasParams.operatorFeeConstant != 0 {
+				rs[i].OperatorFeeScalar = u32ptrTou64ptr(gasParams.operatorFeeScalar)
+				rs[i].OperatorFeeConstant = gasParams.operatorFeeConstant
+			} else {
+				rs[i].OperatorFeeScalar = nil
+				rs[i].OperatorFeeConstant = nil
+			}
```

Comment:
This would only set the fields only if _both_ are non-zero. But I think there are legitimate cases where one of the parameters, especially the operator-fee-constant, are zero.
```suggestion
			if gasParams.operatorFeeScalar != nil && gasParams.operatorFeeConstant != nil && (*gasParams.operatorFeeScalar != 0 || *gasParams.operatorFeeConstant != 0) {
				rs[i].OperatorFeeScalar = u32ptrTou64ptr(gasParams.operatorFeeScalar)
				rs[i].OperatorFeeConstant = gasParams.operatorFeeConstant
			}
```
I think we can also leave out the else branch.

Reply:
Done